src/KM_Controls.pas

```diff
@@ -3424,21 +3459,32 @@ procedure TKMNumericEdit.SetVisible(aValue: Boolean);
 
 function TKMNumericEdit.GetSelfAbsLeft: Integer;
 begin
-  Result := AbsLeft + 21; // Need to add extra +1 because of left button border
+  Result := AbsLeft + 20; // Need to add extra +1 because of left button border
 end;
 
 
 function TKMNumericEdit.GetSelfWidth: Integer;
 begin
-  Result := fWidth - 42; // Need to substruct extra -2 because of 2 buttons borders
+  Result := fWidth - 40; // Need to substruct extra -2 because of 2 buttons borders
+end;
+
+
+function TKMNumericEdit.IsCharValid(Key: WideChar): Boolean;
+begin
+  Result := SysUtils.CharInSet(Key, ['0'..'9']);
+end;
+
+
+procedure TKMNumericEdit.SetValueNCheckRange(aValue: Int64);
+begin
+  SetValue(EnsureRange(aValue, Low(Integer), High(Integer)));

```

Comment:
Allow big numbers in TKMNumericEdit. Fixed edge errors (Integer out of range when incrementing/decr. with one of the possible ways - button/mousewheel/keyboard).
Toxic used big numbers in NumEdits for his RMG settings popup window, so I fixed these small issues